planner/scan.go

```diff
@@ -103,23 +115,32 @@ func (n *scanNode) Next() (bool, error) {
 		return false, nil
 	}
 
-	// keep scanning until we find a doc that passes the filter
-	for {
-		var err error
-		n.docKey, n.currentValue, err = n.fetcher.FetchNextDoc(n.p.ctx, n.documentMapping)
-		if err != nil {
-			return false, err
-		}
-		n.execInfo.docFetches++
+	var err error
+	n.docKey, n.currentValue, err = n.fetcher.FetchNextDoc(n.p.ctx, n.documentMapping)
+	if err != nil {
+		return false, err
+	}
+	// @question: Should this be after the length check?
+	n.execInfo.docFetches++
 
-		if len(n.currentValue.Fields) == 0 {
-			return false, nil
-		}
-		n.documentMapping.SetFirstOfName(
-			&n.currentValue,
-			request.DeletedFieldName,
-			n.currentValue.Status.IsDeleted(),
-		)
+	if len(n.currentValue.Fields) == 0 {
+		return false, nil
+	}
+	n.documentMapping.SetFirstOfName(
+		&n.currentValue,
+		request.DeletedFieldName,
+		n.currentValue.Status.IsDeleted(),
+	)
+
+	// Keeping for refence as there is an outstanding
+	// question.
+	//
+	// Don't need to run filter here anymore since the
+	// fetcher will run the filter for us
+	//
+	// Question: How do we want to handle `info.filterMatches`
+	// now? @shahzadlone
+	/*
```

Comment:
@shahzadlone

Reply:
Technically if `len(n.currentValue.Fields) != 0` shouldn't the state still be the same (i.e. doc match was found)? It's not explicitly a filter match, but would be fine to keep it anywhere after the if and before the `return true, nil`.
Also, open to changing the name if you have any better ideas to represent a "successful document found while scanning / fetching, which matched / passed the filters"